tests/depositWithdrawTest.ts

```diff
@@ -0,0 +1,263 @@
+      console.log("Contract Account Before",balanceOfContract.toString(10));
+      let currentRid = new BN(0);
+      let currentMerkleRoot = "";
+      await proxy.getProxyInfo().then((Proxyinfo:any)=>{
+        currentRid = Proxyinfo.rid;
+        currentMerkleRoot = Proxyinfo.merkle_root.toString();
+      });
+      let tx = proxy.verify(command,[new BN("0")],[new BN("0")],[new BN("0")],[[currentMerkleRoot, currentMerkleRoot, sha_low.toString(), sha_high.toString()]], vid, new BN(currentRid));
+      let r = await tx.when("Verify", "transactionHash", (hash: string) => {
+        console.log("Get transactionHash", hash);
+        txhash = hash;
+      });
+      console.log("done", r.blockHash);
+      let balanceOfMonitorAfter = await tokenContract.balanceOf(config.monitorAccount);
+      console.log("Monitor Account After",balanceOfMonitorAfter.toString(10));
+      let balanceOfContractAfter = await tokenContract.balanceOf(ProxyJSON.networks[config.deviceId].address);
+      console.log("Contract Account After",balanceOfContractAfter.toString(10));
+      if(balanceOfMonitor.toString(10) == balanceOfMonitorAfter.sub(totalAmount).toString(10)){
+        console.log("Monitor Balance Check: PASSED")
+      }else{
+        console.log("Monitor Balance Check: FAILED")
+      }
+      if(balanceOfContractAfter.toString(10) == balanceOfContract.sub(totalAmount).toString(10)){
+        console.log("Contract Balance Check: PASSED")
+      }else{
+        console.log("Contract Balance Check: FAILED")
+      }
+      return r;
+    } catch (e: any) {
+      if (txhash !== "") {
+        console.log("exception with transactionHash ready", " will retry ...");
+        console.log("exception with transactionHash ready", " will retry ...");
+        throw e;
+      } else {
+        if (e.message == "ESOCKETTIMEDOUT") {
+          await new Promise((resolve) => setTimeout(resolve, 5000));
+        } else if (e.message == "nonce too low") {
+          console.log("failed on:", l1client.getChainIdHex(), e.message); // not sure
+          return;
+        } else {
+          console.log("Unhandled exception during verify");
+          console.log(e);
+          throw e;
+        }
+      }
+    }
+  }
+}
+
+async function main() {
+    let testChain = process.argv[2];
+    let pendingEvents: [Field, Field[]][] = [];
+    console.log(
+        "========================== Deposit & Withdraw Balance Check =========================="
+        );
+      await mintToken(testChain);
+      let tokenIdx = await addToken(testChain);
+    function dataToBN(data: any) {
+      if (data.toHex) {
+        data = data.toHex();
+      }
+      return new BN(data, 16);
+    }
+    let nonce = 0;
+    let accountIndex = 0;
+    let tokenIndex = tokenIdx;
+    let amount = 1;
+    let config = await getConfigByChainName(L1ClientRole.Monitor, testChain);
+    let l1address = encodeL1address("4D9A852e6AECD3A6E87FecE2cA109780E45E6F2D", parseInt(config.deviceId).toString(16));
```

Comment:
why we have hard code l1 address here?
I can see in the mintToken function you use config.monitorAccount. So here should not be hardcode?